test/runner.py

```diff
@@ -563,6 +567,94 @@ def cleanup_emscripten_temp():
         pass
 
 
+def print_repository_info(directory, repository_name):
+  import subprocess
+  current_commit = subprocess.run(['git', 'log', '-n1'], cwd=directory, stdout=subprocess.PIPE, text=True).stdout.strip()
+  print(f'\n{repository_name} {current_commit}\n')
+  local_changes = subprocess.run(['git', 'diff'], cwd=directory, stdout=subprocess.PIPE, text=True).stdout.strip()
+  if local_changes:
+    print(f'\n{local_changes}\n')
+
+
+def log_test_environment():
+  """Print detailed information about the current test environment. Useful for
+  logging test run configuration in a CI."""
+  print('======================== Test Setup ========================')
+  print(f'Test time: {datetime.datetime.now(datetime.timezone.utc).strftime("%A, %B %d, %Y %H:%M:%S %Z")}')
+  print(f'Python: "{sys.executable}". Version: {sys.version}')
+  print(f'Emscripten test runner path: "{os.path.realpath(__file__)}"')
+
+  if os.path.isdir(os.path.join(config.EMSCRIPTEN_ROOT, '.git')):
+    print(f'\nEmscripten repository: "{config.EMSCRIPTEN_ROOT}"')
+
+  emscripten_version = os.path.realpath(os.path.join(config.EMSCRIPTEN_ROOT, 'emscripten-version.txt'))
+  if os.path.isfile(emscripten_version):
+    print(f'emscripten-version.txt: {utils.read_file(emscripten_version).strip()}')
```

Comment:
You can just use `utils.EMSCRIPTEN_VERSION` here (its is populated from this file)